ang/civicase/case/search/directives/search.directive.js

```diff
@@ -266,6 +266,28 @@
       }
     }
 
+    /**
+     * Returns case types filtered by given category
+     *
+     * @param {String} categoryName
+     * @return {Array}
+     */
+    function getCaseTypesFilteredByCategory (categoryName) {
+      var filteredCaseTypes = [];
+
+      var caseTypeCategory = _.find(caseTypeCategories, function (category) {
+        return category.name.toLowerCase() === $scope.filters.case_type_category.toLowerCase();
+      });
+
+      if (caseTypeCategory) {
+        filteredCaseTypes = _.filter(caseTypes, function (caseType) {
+          return caseType.case_type_category === caseTypeCategory.value;
+        });
+      }
```

Comment:
Lets exist the function early instead of declaring an extra var:

```js
function getCaseTypesFilteredByCategory (categoryName) {
  var caseTypeCategory = _.find(caseTypeCategories, function (category) {
    return category.name.toLowerCase() === $scope.filters.case_type_category.toLowerCase();
  });

  if (!caseTypeCategory) {
    return [];
  }

  return _.filter(caseTypes, function (caseType) {
    return caseType.case_type_category === caseTypeCategory.value;
  });
}
```